File: modele_classes_java_simple/Invoice.java
import java.util.Date;
import java.util.List;

/**
 * Facture client
 */
public class Invoice {
    
    // Attributs
    private String id;
    private String numeroFacture;
    private Date periodeDebut;
    private Date periodeFin;
    private double montantTtc;
    private StatutFacture statut;
    private String fichierPdf;
    
    // Relations
    private Company company;
    private Line line;
    private List<NotificationFacture> notifications;
    private List<HistoriqueFacturation> historique;
    
    // Méthodes
    public void publier() {
        this.statut = StatutFacture.PUBLIEE;
    }
    
    public void annuler() {
        this.statut = StatutFacture.ANNULEE;
    }
}

File: modele_classes_csharp/Invoice.cs
using System;
using System.Collections.Generic;
using Moov.Facturation.Contrats;

namespace Moov.Facturation.Facturation
{
    /// <summary>
    /// Facture client (globale ou individuelle)
    /// </summary>
    public class Invoice
    {
        // Attributs
        public string Id { get; set; }
        public string NumeroFacture { get; set; } // unique
        public DateTime PeriodeDebut { get; set; }
        public DateTime PeriodeFin { get; set; }
        public double MontantTtc { get; set; }
        public StatutFacture Statut { get; set; }
        public string FichierPdf { get; set; }
        
        // Relations
        public Company Company { get; set; }
        public Line Line { get; set; } // nullable pour facture globale
        public List<NotificationFacture> Notifications { get; set; }
        public List<HistoriqueFacturation> Historique { get; set; }
        
        // Constructeur
        public Invoice()
        {
            Notifications = new List<NotificationFacture>();
            Historique = new List<HistoriqueFacturation>();
        }
        
        // Méthodes métier
        /// <summary>
        /// Publie la facture
        /// </summary>
        public void Publier()
        {
            Statut = StatutFacture.PUBLIEE;
        }
        
        /// <summary>
        /// Annule la facture
        /// </summary>
        public void Annuler()
        {
            Statut = StatutFacture.ANNULEE;
        }
    }
}
